src/nodes.coffee

```diff
@@ -1695,14 +1713,38 @@ exports.CSXElement = class CSXElement extends Base
 
     {openingFragment, closingFragment}
 
+  contentAst: (o) ->
```

Comment:
Took the approach of having the `CSXElement` parent be responsible (during AST generation) for extracting the relevant individual children and eg instantiating new `CSXText`/`CSXEmptyExpression` node classes (only used currently for AST generation)

Reply:
Which feels like it should be the job of the lexer/parser, right? But refactoring to make that happen is probably way too big a scope.

Reply:
Yup, with JSX the options are:

1. Fully lex/parse as JSX rather than passing JSX through the grammar as pseudo-calls etc. We've discussed that this might ultimately be worth doing (it'd avoid some of the backflips we've done in these PRs to eg preserve JSX location data) but isn't something we're considering as in scope for this AST work

So then in order to avoid either manually assembling JSX AST nodes or having node classes like `Call` serve "double-duty" at AST generation time and have their `ast()` method check whether they're actually a JSX node, we've been either

2. Converting to JSX-specific node class instances "up front" (ie during node class construction), so that the JSX-specific node classes are responsible for both compile-to-JS (ie have `compileNode()` methods) and AST generation. This is what we've done with eg `CSXElement`

3. Converting to JSX-specific node class instances during AST generation (leaving the existing compile-to-JS code as-is)

In this case I took # 3, since it felt like it could be cleanly handled by the `CSXElement` parent without having to mess with the existing code in `StringWithInterpolations::compileNode()` for differentiating JSX content from actual interpolated string contents